src/Tribe/Values/Abstract_Value.php

```diff
@@ -0,0 +1,455 @@
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed
+	 */
+	public function get_initial_representation() {
+		return $this->initial_value;
+	}
+
+	/**
+	 * Transforms any formatted numeric string into a numeric value
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value the formatted string.
+	 *
+	 * @return float
+	 */
+	public function normalize( $value ) {
+
+		if ( is_numeric( $value ) ) {
+			/**
+			 * Filter the value returned for normalize() when implemented in a specific class name
+			 *
+			 * @since TBD
+			 *
+			 * @param float $value the normalized value
+			 * @param Abstract_Value the object instance
+			 *
+			 * @return float
+			 */
+			$value = (float) apply_filters( "tec_common_value_{$this->class_name}_normalized", (float) $value, $this );
+
+			/**
+			 * Filter the value returned for normalize() when implemented in a specific class name
+			 *
+			 * @since TBD
+			 *
+			 * @param float $value the normalized value
+			 * @param Abstract_Value the object instance
+			 *
+			 * @return float
+			 */
+			do_action( 'tec_common_value_normalize_is_numeric', $value, $this );
+
+			/**
+			 * Filter the value returned for get_precision() when implemented in any class
+			 *
+			 * @since TBD
+			 *
+			 * @param float $value the normalized value
+			 * @param Abstract_Value the object instance
+			 *
+			 * @return float
+			 */
+			return (float) apply_filters( 'tec_common_value_normalized', $value, $this );
+		}
+
+		if ( $this->is_character_block( $value ) ) {
+			return (float) 0;
+		}
+
+		// If we can split the amount by spaces, remove any blocks that don't contain any digits
+		// This is important in case the currency unit contains the same characters as the decimal/thousands
+		// separators such as in Moroccan Dirham (1,234.56 .د.م.) or Danish Krone (kr. 1.234,56)
+		foreach ( explode( ' ', $value ) as $block ) {
+			if ( $this->is_character_block( $block ) ) {
+				$value = str_replace( $block, '', $value );
+			}
+		}
+
+		// Remove encoded html entities
+		$value = preg_replace( '/&[^;]+;/', '', trim( $value ) );
+
+		// Get all non-digits from the amount
+		preg_match_all( '/[^\d]/', $value, $matches );
+
+		// if the string is all digits, it is numeric
+		if ( empty( $matches[0] ) ) {
+			return (float) $value;
+		}
+
+		$tokens    = array_unique( $matches[0] );
+		$separator = '/////';
+
+		foreach ( $tokens as $token ) {
+			if ( $this->is_decimal_separator( $token, $value ) ) {
+				$separator = $token;
+				continue;
+			}
+
+			$value = str_replace( $token, '', $value );
+		}
+
+		$pieces = explode( $separator, $value );
+
+		// If the initial amount did not have decimals specified, $pieces will be an array of a single
+		// numeric value, so we just return it as a float.
+		if ( 1 === count( $pieces ) && is_numeric( reset( $pieces ) ) ) {
+			return (float) reset( $pieces );
+		}
+
+		$decimal = array_pop( $pieces );
+
+		return (float) implode( '', array_merge( $pieces, [ '.', $decimal ] ) );
```

Comment:
I think this whole piece here deserves it's own method, since it has a ton of small complexities that we might want to call separately.

Reply:
@bordoni not sure I understand. Do you want me to break down `normalize` into a sequence of smaller methods? I can do that.

Reply:
Or do you want to make it a separate class?